llvm/lib/Target/AArch64/AArch64TargetObjectFile.cpp

```diff
@@ -28,6 +29,21 @@ void AArch64_ELFTargetObjectFile::Initialize(MCContext &Ctx,
   SupportDebugThreadLocalLocation = false;
 }
 
+void AArch64_ELFTargetObjectFile::emitPersonalityValueImpl(
+    MCStreamer &Streamer, const DataLayout &DL, const MCSymbol *Sym,
+    const MachineModuleInfo *MMI) const {
+  if (!MMI->getObjFileInfo<MachineModuleInfoELF>().hasSignedPersonality()) {
```

Comment:
We should add a subclass of MachineModuleInfoELF similar to `AMDGPUMachineModuleInfo`, then move `HasSignedpersonality` there.  Here you can use a static_cast.

Reply:
> We should add a subclass of MachineModuleInfoELF similar to `AMDGPUMachineModuleInfo`, then move `HasSignedpersonality` there. Here you can use a static_cast.

Thanks for suggestion, please let me know if b4ca8b141b3df8737f0f0d6f64f083fefb403735 resolves your comment